lib/spack/spack/cmd/diy.py

```diff
@@ -89,14 +86,45 @@ def diy(self, args):
     package.stage = DIYStage(source_path)
 
     # disable checksumming if requested
-    if args.no_checksum:
+    if args.no_checksum or args.yes_to_all:
         spack.config.set('config:checksum', False, scope='command_line')
 
-    package.do_install(
+    kwargs = dict(
         make_jobs=args.jobs,
         keep_prefix=args.keep_prefix,
         install_deps=not args.ignore_deps,
         verbose=not args.quiet,
         keep_stage=True,   # don't remove source dir for DIY.
-        tests=args.run_tests,
+        tests=tests,
         dirty=args.dirty)
+
+    if args.overwrite:
+        assert package.installed, "to overwrite a spec you must install it first"
```

Comment:
In the upstream they now allow non-installed spec with overwrite: https://github.com/spack/spack/pull/9201